utils/src/test/scala/za/co/absa/enceladus/utils/general/SectionSuite.scala

```diff
@@ -0,0 +1,700 @@
+  }
+
+  test("ofSameChars: negative fromIndex") {
+    val section = Section.ofSameChars("xxxxyyyzz", -5)
+    val expected = Section(-5, 3)
+    assert(section == expected)
+  }
+
+  test("ofSameChars: negative fromIndex far ahead") {
+    val section = Section.ofSameChars("xxxxyyyzz", -15)
+    val expected = Section(-15, 0)
+    assert(section == expected)
+  }
+
+  //removeMultiple
+  test("removeMultiple: two Sections") {
+    val sections = Seq(Section(2, 2), Section(6, 3))
+    val result = Section.removeMultiple("abcdefghijkl", sections)
+    val expected = "abefjkl"
+    assert(result == expected)
+  }
+
+  test("removeMultiple: three Sections, unordered") {
+    val sections = Seq(Section(6, 3), Section(2, 2), Section(11, 2))
+    val result = Section.removeMultiple("abcdefghijklmnop", sections)
+    val expected = "abefjknop"
+    assert(result == expected)
+  }
+
+  test("removeMultiple: adjacent") {
+    val sections = Seq(Section(2, 2), Section(11, 2), Section(6, 5))
+    val result = Section.removeMultiple("abcdefghijklmnop", sections)
+    val expected = "abefnop"
+    assert(result == expected)
+  }
+
+  test("removeMultiple: overlapping") {
+    val sections = Seq(Section(2, 2), Section(10, 3), Section(6, 5))
+    val result = Section.removeMultiple("abcdefghijklmnop", sections)
+    val expected = "abefnop"
+    assert(result == expected)
+  }
+
+  test("removeMultiple: negative") {
+    val sections = Seq(Section(1, 1), Section(-2, 1))
+    val result = Section.removeMultiple("abcdefghijklmnop", sections)
+    val expected = "acdefghijklmnp"
+    assert(result == expected)
+  }
+
+  test("removeMultiple: negative overlapping") {
+    val sections = Seq(Section(-3, 3), Section(-5, 3))
+    val result = Section.removeMultiple("abcdefghijklmnop", sections)
+    val expected = "abcdefghijk"
+    assert(result == expected)
+  }
+
+  test("removeMultiple: running out of bounds ") {
+    val sections = Seq(Section(-6, 2), Section(4, 2))
+    val result = Section.removeMultiple("abcde", sections)
+    val expected = "bcd"
+    assert(result == expected)
+  }
+
+  test("removeMultiple: totally out of bounds ") {
+    val sections = Seq(Section(-10, 2), Section(10, 2))
+    val result = Section.removeMultiple("abcde", sections)
+    val expected = "abcde"
+    assert(result == expected)
+  }
+
+  test("removeMultiple: zero length") {
+    val sections = Seq(Section(1, 0), Section(-2, 0))
+    val result = Section.removeMultiple("abcdefghijklmnop", sections)
+    val expected = "abcdefghijklmnop"
+    assert(result == expected)
+  }
+
+  //mergeSections
+  test("mergeSections: empty") {
+    val sections = Seq.empty[Section]
+    val result = Section.mergeSections(sections)
+    assert(result.isEmpty)
+  }
+
+  test("mergeSections: one item") {
+    val sections = Seq(Section(42, 7))
+    val result = Section.mergeSections(sections)
+    assert(result == sections)
+  }
+
+  test("mergeSections: touching") {
```

Comment:
Maybe illustrate an example of to with the explanation why for the given input we get a particular output?

For instance,
```
For a string:
1234567890ACDFEFGHIJKLMNOPQUSTUVWXYZ
 ^ ^                            ^^
 | | ^^                         |
 | | Section(5,1)               Section(-4,2)
 | Section(3,2)
 Section(1,1)

Output of the merge:
...
```
etc.

That is so we won't forget your explanation.